src/main/kotlin/MyPackageManager.kt

```diff
@@ -21,33 +21,35 @@ package org.ossreviewtoolkit.plugins.packagemanagers.mypackagemanager
 
 import java.io.File
 
-import org.ossreviewtoolkit.analyzer.AbstractPackageManagerFactory
 import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.analyzer.PackageManagerFactory
 import org.ossreviewtoolkit.model.Project
 import org.ossreviewtoolkit.model.ProjectAnalyzerResult
 import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
-import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.config.Excludes
 import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.plugins.api.OrtPlugin
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
 
+@OrtPlugin(
+    displayName = "My Package Manager",
+    description = "A template for implementing an ORT package manager plugin.",
+    factory = PackageManagerFactory::class
+)
 class MyPackageManager(
-    name: String,
-    analysisRoot: File,
-    analyzerConfig: AnalyzerConfiguration,
-    repoConfig: RepositoryConfiguration
-) : PackageManager(name, "MyProjectType", analysisRoot, analyzerConfig, repoConfig) {
-    class Factory : AbstractPackageManagerFactory<MyPackageManager>("MyPackageManager") {
-        override val globsForDefinitionFiles = listOf("MyPackageManager.DefinitionFile")
+    override val descriptor: PluginDescriptor = MyPackageManagerFactory.descriptor
+) : PackageManager("MyProjectType") {
+    override val globsForDefinitionFiles = listOf("MyPackageManager.DefinitionFile")
 
-        override fun create(
-            analysisRoot: File,
-            analyzerConfig: AnalyzerConfiguration,
-            repoConfig: RepositoryConfiguration
-        ) = MyPackageManager(type, analysisRoot, analyzerConfig, repoConfig)
-    }
-
-    override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
+    override fun resolveDependencies(
+        analysisRoot: File,
+        definitionFile: File,
+        excludes: Excludes,
+        analyzerConfig: AnalyzerConfiguration,
+        labels: Map<String, String>
+    ): List<ProjectAnalyzerResult> {
         val issue = createAndLogIssue(
-            source = managerName,
+            source = MyPackageManagerFactory.descriptor.id,
```

Comment:
That should be the `displayName` instead of the `id`, but actually with the new version of that helper function you could just use `createAndLogIssue("Not yet implemented.")` and the `displayName` will be used automatically as source.